Nodify/Connections/BaseConnection.cs

```diff
@@ -118,6 +118,7 @@ public abstract class BaseConnection : Shape
         public static readonly DependencyProperty SourceOffsetModeProperty = DependencyProperty.Register(nameof(SourceOffsetMode), typeof(ConnectionOffsetMode), typeof(BaseConnection), new FrameworkPropertyMetadata(ConnectionOffsetMode.Static, FrameworkPropertyMetadataOptions.AffectsRender));
         public static readonly DependencyProperty TargetOffsetModeProperty = DependencyProperty.Register(nameof(TargetOffsetMode), typeof(ConnectionOffsetMode), typeof(BaseConnection), new FrameworkPropertyMetadata(ConnectionOffsetMode.Static, FrameworkPropertyMetadataOptions.AffectsRender));
         public static readonly DependencyProperty DirectionProperty = DependencyProperty.Register(nameof(Direction), typeof(ConnectionDirection), typeof(BaseConnection), new FrameworkPropertyMetadata(default(ConnectionDirection), FrameworkPropertyMetadataOptions.AffectsRender));
+        public static readonly DependencyProperty OrientationProperty = StackPanel.OrientationProperty.AddOwner(typeof(BaseConnection), new FrameworkPropertyMetadata(Orientation.Horizontal));
```

Comment:
As discussed i recommend the split for more flexiblity

Reply:
Thanks for implementing this. We can now draw all kinds of combinations:

![image](https://github.com/miroiu/nodify/assets/12727904/4be22599-9644-4ad2-92a0-be1b7e246f23)

Reply:
im kind of curios how you placed your start/end here so it aligns beneath each other smoothly, i based my xaml on the playground and i had to create a attached property which modifies the header/footer port padding, and a visibility binding on those two ports to remove the title.

Did you create a kind of redundant NodeInput/NodeOutput copy of the node class/style to achieve this? Or was there a simpler way avoiding the copy + what i described.

Reply:
I used a simple connector which does not have any padding:

```xaml
<nodify:Node.HeaderTemplate>
    <DataTemplate>
        <nodify:Connector Anchor="{Binding Anchor, Mode=OneWayToSource}" IsConnected="{Binding IsConnected}" />
    </DataTemplate>
</nodify:Node.HeaderTemplate>
```

However, the default `Connector` style does not have a hover effect. You may also want to include a trigger for `IsMouseOver` in the `ControlTemplate.Triggers` collection:

```xaml
<Trigger Property="IsMouseOver"
         Value="True">
    <Setter TargetName="Connector"
            Property="Ellipse.Fill"
            Value="{Binding BorderBrush, RelativeSource={RelativeSource TemplatedParent}}" />
</Trigger>
```